tests/config/test_omegaconf_config.py

```diff
@@ -487,3 +488,20 @@ def test_env_resolver_is_registered_after_loading(self, tmp_path):
         assert conf["credentials"]["user"]["name"] == "test_user"
         assert OmegaConf.has_resolver("oc.env")
         OmegaConf.clear_resolver("oc.env")
+
+    @use_config_dir
+    def test_load_config_from_tar_file(self, tmp_path):
+        subprocess.run(  # pylint: disable=subprocess-run-check
+            [
+                "tar",
+                "--exclude=local/*.yml",
+                "-czf",
+                f"{tmp_path}/tar_conf.tar.gz",
+                f"--directory={str(tmp_path.parent)}",
+                f"{tmp_path.name}",
+            ]
+        )
+
+        conf = OmegaConfigLoader(conf_source=f"{tmp_path}/tar_conf.tar.gz")
+        catalog = conf["catalog"]
+        assert catalog["trains"]["type"] == "MemoryDataSet"
```

Comment:
(Y) It would be great to have another test for `zip` too?

Reply:
As I mentioned in the development notes it's really tricky to create a `zip` file with the correct structure when working with `tmp_path`.. 😕 I'll see if I can do something similar to what fsspec is doing.

Reply:
Oh actually I found a way! I'll add the test.